google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,303 @@
+    this.project = project;
+    this.gcpProjectId = gcpProjectId;
+    this.downloadPath = downloadPath;
+    this.configurationSettingsList = configurationSettingsList;
+
+    tableModel =
+        new BooleanTableModel<>(
+            configurationSettingsList,
+            RunnerAndConfigurationSettings.class,
+            Comparator.comparing(RunnerAndConfigurationSettings::getName),
+            true);
+
+    init();
+    setTitle(GctBundle.message("cloud.apis.service.account.key.downloaded.title"));
+    table.setTableHeader(null);
+
+    if (tableModel.getRowCount() == 0) {
+      label.setVisible(false);
+      scrollPane.setVisible(false);
+      table.setVisible(false);
+    }
+  }
+
+  @Nullable
+  @Override
+  protected JComponent createCenterPanel() {
+    return mainPanel;
+  }
+
+  @NotNull
+  @Override
+  protected Action[] createActions() {
+    List<Action> actions = new ArrayList<>();
+    actions.add(getOKAction());
+    if (tableModel.getRowCount() > 0) {
+      actions.add(new ApplyAction());
+    }
+    return actions.toArray(new Action[0]);
+  }
+
+  private void createUIComponents() {
+    commonPanel = new ServiceAccountKeyDownloadedPanel(gcpProjectId, downloadPath);
+
+    if (tableModel == null) {
+      List<RunnerAndConfigurationSettings> configurationSettingsList =
+          getAppEngineStandardConfigurationSettingsList();
+      tableModel =
+          new BooleanTableModel<>(
+              configurationSettingsList,
+              RunnerAndConfigurationSettings.class,
+              Comparator.comparing(RunnerAndConfigurationSettings::getName),
+              true);
+    }
+    table = new ServerTable(tableModel);
+  }
+
+  private List<RunnerAndConfigurationSettings> getAppEngineStandardConfigurationSettingsList() {
+    if (configurationSettingsList != null) {
+      return configurationSettingsList;
+    }
+
+    return RunManager.getInstance(project)
+        .getConfigurationSettingsList(AppEngineServerConfigurationType.getInstance());
+  }
+
+  private Set<RunnerAndConfigurationSettings> getSelectedConfigurations() {
+    return tableModel.getSelectedItems();
+  }
+
+  @VisibleForTesting
+  public void addEnvironmentVariablesToConfiguration(
+      Set<RunnerAndConfigurationSettings> configurations) {
+    String executorId = DefaultRunExecutor.getRunExecutorInstance().getId();
+    configurations.forEach(
+        configurationSettings ->
+            addEnvironmentVariablesToConfiguration(executorId, configurationSettings));
+  }
+
+  /**
+   * Adds the environment variables for the Google Cloud Libraries to the list of environment
+   * variables for {@code configuration} if they don't exist. If they exist, it replaces them.
+   */
+  private void addEnvironmentVariablesToConfiguration(
+      String executorId, RunnerAndConfigurationSettings configuration) {
+    ProgramRunner runner = ProgramRunnerUtil.getRunner(executorId, configuration);
+    if (runner == null) {
+      LOG.error(
```

Comment:
Logging errors bring up pretty harsh plugin level error window, could this be a bit too much here? How about showing this as a error in this dialog itself (I think `DialogWrapper` supports error messages) so a user clearly sees what happened and log WARN message instead?